ClosedXML/Excel/CalcEngine/Functions/Text.cs

```diff
@@ -275,6 +276,61 @@ private static object _Text(List<Expression> p)
             else
                 return nf.Format(number, CultureInfo.InvariantCulture);
         }
+        /// <summary>
+        /// A function to Join text https://support.office.com/en-us/article/textjoin-function-357b449a-ec91-49d0-80c3-0e8fc845691c
+        /// </summary>
+        /// <param name="p">Parameters</param>
+        /// <returns> string </returns>
+        /// <exception cref="ApplicationException">
+        /// Delimiter in first param must be a string
+        /// or
+        /// Second param must be a boolen (TRUE/FALSE)
+        /// </exception>
+        private static object TextJoin(List<Expression> p)
+        {
+            List<string> values = new List<string>();
+            var p0 = p[0].Evaluate();
+            var p1 = p[1].Evaluate();
+
+            if (!(p0 is string)) throw new ApplicationException("Delimiter in first param must be a string");
```

Comment:
I don't think we should be throwing `ApplicationException` here. What does Excel return when you input a non-string value in that parameter?